src/core/annotation.js

```diff
@@ -3767,37 +3774,50 @@ class FreeTextAnnotation extends MarkupAnnotation {
     const { evaluatorOptions, xref } = params;
     this.data.annotationType = AnnotationType.FREETEXT;
     this.setDefaultAppearance(params);
-    if (this.appearance) {
+    this._hasAppearance = !!this.appearance;
+
+    if (this._hasAppearance) {
       const { fontColor, fontSize } = parseAppearanceStream(
         this.appearance,
         evaluatorOptions,
         xref
       );
       this.data.defaultAppearanceData.fontColor = fontColor;
       this.data.defaultAppearanceData.fontSize = fontSize || 10;
-    } else if (this._isOffscreenCanvasSupported) {
-      const strokeAlpha = params.dict.get("CA");
-      const fakeUnicodeFont = new FakeUnicodeFont(xref, "sans-serif");
-      this.data.defaultAppearanceData.fontSize ||= 10;
-      const { fontColor, fontSize } = this.data.defaultAppearanceData;
-      this.appearance = fakeUnicodeFont.createAppearance(
-        this._contents.str,
-        this.rectangle,
-        this.rotation,
-        fontSize,
-        fontColor,
-        strokeAlpha
-      );
-      this._streams.push(this.appearance, FakeUnicodeFont.toUnicodeStream);
     } else {
-      warn(
-        "FreeTextAnnotation: OffscreenCanvas is not supported, annotation may not render correctly."
-      );
+      const { fontColor, fontSize } = this.data.defaultAppearanceData;
```

Comment:
Could `fontSize` not potentially be zero here, in which case I'm not sure if `FakeUnicodeFont.getFirstPositionInfo` will return meaningful data?

Looking at the old code we used to do `this.data.defaultAppearanceData.fontSize ||= 10;` *before* getting the `fontColor`/`fontSize` so should we not keep that behaviour (by moving that line) to avoid any issues?
```suggestion
      this.data.defaultAppearanceData.fontSize ||= 10;
      const { fontColor, fontSize } = this.data.defaultAppearanceData;
```